riscv-c-api.md

```diff
@@ -48,10 +48,41 @@ https://creativecommons.org/licenses/by/4.0/.
 | __riscv_fsqrt       | 1     | `F` extension is available and `-mno-fdiv` is not given.*[1]   |
 | __riscv_vector      | 1     | `V` extension is available.   |
 | __riscv_bitmanip    | 1     | `B` extension is available.   |
-| __riscv_zfh         | 1     | `Zfh` extension is available. |
 
 *[1] Not all compilers provide `-mno-div` and `-mno-fdiv` option.
 
+### Architecture Extension Test Macro
+
+Architecture extension test macro is a new set of test macro to checking the
+availability and version for certain extension, however not all compilers are
+supported, so you should check `__riscv_arch_test_macro` to make sure this
+compiler is supporting those preprocessor definitions.
+
+The value of architecture extension test macro are defined as its version,
+which is compute by following formula:
+
+```
+<MAJOR_VERSION> * 1,000,000 + <MINOR_VERSION> * 1,000 + <REVISION_VERSION>
+```
+
+For example:
+- F-extension v2.2 will define `__riscv_f` as `20002000`.
+- B-extension v0.92 will define `__riscv_b` as `92000`.
+
+| Name                    | Value        | When defined                  |
+| ----------------------- | ------------ | ----------------------------- |
+| __riscv_arch_test_macro | 1            | Defined if compiler support new architecture extension test macro. |
```

Comment:
Delete this line.

Reply:
It's a macro to indicate this compiler has supported those macro. 
e.g.
```
#ifdef __riscv_arch_test_macro
// We can assume __riscv_i and __riscv_<ext> is supported on this compiler.
#endif
```

Reply:
The intention of this macro is I believe there would be an junction period, some compiler provide those macro, and some compiler not ready yet, and this macro can tell user it's supported or not.

Reply:
Methinks that it's not necessary.  The respective macros can be tested for `#ifdef`.

Reply:
I would prefer keep this macro, if user only want to use new macro, they can have way to checking compiler is supported, and maybe emit error when compiler not supported, without the macro we need to tested that by `#if defined(__riscv_i) || defined(__riscv_e)`.

```
#ifndef __riscv_arch_test_macro
#error Compiler too old, plz upgrade your RISC-V compiler.
#endif
```

Reply:
Fair enough.  However, a macro doesn't have to be named `..._macro`.

Reply:
Fixed